r 3.3.0

erichare opened the pull request:
### Have you:

- [x] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [x] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [x] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [x] Built your formula locally prior to submission with `brew install <formula>`?

### New formula: have you

- [ ] Written a sensible test? (The best test is to compile and run a sample program.)

### Updates to existing formula: have you

- [x] Removed the `revision` line, if any, when bumping the version number?
- [x] Added/bumped the `revision` number if the changes affect the precompiled bottles?
- [x] Not altered the `bottle` section?
- [x] Checked that the tests still pass?

tdsmith commented on r.rb:
```diff
@@ -179,8 +182,8 @@ def caveats
         R CMD javareconf JAVA_CPPFLAGS="-I/System/Library/Frameworks/JavaVM.framework/Headers -I/Library/Java/JavaVirtualMachines/jdk<version>.jdk/"
-      (where <version> can be found by running `java -version`, `/usr/libexec/java#{'_'}home`, or `locate jni.h`), or:
-        R CMD javareconf JAVA_CPPFLAGS="-I/System/Library/Frameworks/JavaVM.framework/Headers -I$(/usr/libexec/java#{'_'}home | grep -o '.*jdk')"
+      (where <version> can be found by running `java -version`, `/usr/libexec/java_home`, or `locate jni.h`), or:
```
these were naughty but intentional to avoid an annoying audit failure :)

tdsmith commented on r.rb:
```diff
@@ -46,6 +48,7 @@ class R < Formula
   depends_on "pango" => :optional
   depends_on "valgrind" => :optional
   depends_on :x11 => (OS.mac? ? :optional : :recommended)
+  depends_on :java => :optional
```
Unless this changes the build, we shouldn't add it.

tdsmith commented:
Since the minor version of R changed, most of the things that `brew uses r` knows about need to updated; the list is:

```
homebrew/science/kmergenie               homebrew/science/poretools             
homebrew/science/littler                 homebrew/science/quake                 
homebrew/science/nonpareil               homebrew/science/r-gui                 
homebrew/science/plr                     homebrew/science/rstudio-server
```

Can you add commits to this PR that increment the `revision` on the Formula body? If a formula doesn't already have a revision, adding `revision 1` above the bottle blocks would be great.

erichare commented:
Thanks for providing the list! i'll increment the revision on these formulas momentarily.

erichare commented:
I've updated the formulas - I thought I removed the java dependency line and the path issue, but i think it didn't get reflected in the commit. If you want me to push a new commit to fix those two issues again just let me know.

erichare commented:
Because I made some mistakes on this one (no separate commit per updated formula, some changes to the R formula that shouldnt have been made) i opened [Pull Request 3630](https://github.com/Homebrew/homebrew-science/pull/3630) to clean it up. It still has the same error. Feel free to close it and reopen this one if preferred.
